src/profile-logic/tracks.js

```diff
@@ -932,9 +1029,36 @@ export function computeDefaultVisibleThreads(
     return new Set(profile.meta.initialVisibleThreads);
   }
 
+  const allTrackThreads = new Set();
```

Comment:
optional super nit: if we're keeping it now, I'd move the computation of allTrackThreads to a separate function, so that this function is more readable (and easily memoizable too).